File: src/ex_26_Wrapper_Class/Lab204_Wrapper.java
package ex_26_Wrapper_Class;

public class Lab204_Wrapper {
    public static void main(String[] args) {
        int a= 10;
        Integer b = a;
        // Boxing - primitive - Wrapper - AutoBoxing - JVM will do it.
        System.out.println(b.intValue());
        System.out.println(b);
        System.out.println(a);

        // UnBoxing (wrapper -> primitive)
        Integer aa = 43;
        int a1 = aa; // UNBOXING
        System.out.println(a1);

    }
}

File: javascript/ex_26_Wrapper_Class/Lab204_Wrapper.js
// NOTE: JavaScript has no wrapper classes. Primitives are auto-boxed when methods are called on them.
// Boxing and Unboxing do not exist in JavaScript - all number values are just 'number' type.

// Main
let a = 10;
let b = a;
// Boxing - primitive -> Wrapper - AutoBoxing - In JS, this is just a simple assignment
console.log(b);       // .intValue() not needed in JS
console.log(b);
console.log(a);

// UnBoxing (wrapper -> primitive) - In JS, this is just a simple assignment
let aa = 43;
let a1 = aa;          // UNBOXING - no distinction in JS
console.log(a1);
